build_runner/lib/src/entrypoint/options.dart

```diff
@@ -390,3 +411,44 @@ Please update your dev_dependencies section of your pubspec.yaml:
 ''');
   }
 }
+
+Map<String, Map<String, dynamic>> _parseBuilderConfigOverrides(
+    dynamic parsedArg) {
+  final builderConfigOverrides = <String, Map<String, dynamic>>{};
+  if (parsedArg == null) return builderConfigOverrides;
+  var allArgs = parsedArg is List<String> ? parsedArg : [parsedArg as String];
+  for (final define in allArgs) {
+    final parts = define.split('=');
+    const expectedFormat = '--define "<builder_key>=<option>=<value>"';
```

Comment:
Should we support arguments in the form `<builder_key>=<option1>=<value1>,<option2>=<value2>`?

Reply:
We could add support for that later on but I don't want to have to try and deal with it right now, that would significantly complicate the parsing logic because you can provide JSON strings for the values (which are likely to have commas as well).